src/components/widgets/utils.tsx

```diff
@@ -0,0 +1,154 @@
+import React from 'react';
+import { DataType, ICurrencyData, IData, IDataStyle, ITextData } from './types';
+import { Text } from '../../library';
+import BigNumber from 'bignumber.js';
+import { formatNumber } from '../../core/utils/format-number';
+import { normalize, normalizeFontAndLineHeight } from '../../styles/dimensions';
+import { formatTranslate } from '../../core/i18n';
+
+const formatTextData = (data: ITextData): string => {
+    let text = '';
+
+    if (data?.params) {
+        // Translated text
+        const translateKeys = {};
+
+        for (const param of Object.keys(data.params)) {
+            const val = data.params[param];
+
+            if (val.type === DataType.TEXT) {
+                translateKeys[param] = formatTextData(val.data as ITextData);
+            }
+
+            if (val.type === DataType.CURRENCY) {
+                translateKeys[param] = formatCurrencyData(val.data as ICurrencyData);
+            }
+        }
+
+        text = formatTranslate(String(data.value), translateKeys);
+    } else {
+        // Static text
+        text = String(data.value);
+    }
+
+    return text;
+};
+
+const amountFromStd = (value: BigNumber | number | string, decimals: number): BigNumber => {
+    return new BigNumber(value).dividedBy(new BigNumber(10).pow(decimals));
+};
+
+const formatCurrencyData = (data: ICurrencyData): string => {
+    let text = '';
+    let value = data.value;
+
+    if (data?.decimals) {
+        value = amountFromStd(value, data.decimals).toFixed();
+    }
+
+    text = formatNumber(Number(value), {
```

Comment:
before calling `formatNumber` you should do the rounding, with the specific flag (ROUND_UP or ROUND_DOWN). 
then in formatNumber you could use `value.toNumber()`